File: swagger/models/get_classes_request.py
from .base_model import BaseModel


class GetClassesRequest(BaseModel):
	"""
	no description available

	Attributes:

		class_description_ids
		The requested class description IDs.

		class_ids
		The requested class IDs.

		staff_ids
		The requested IDs of the teaching staff members.

		start_date_time
		The requested start date for filtering.   <br />Default: **today’s
		date**

		end_date_time
		The requested end date for filtering.  <br />Default: **today’s
		date**

		client_id
		The client ID of the client who is viewing this class list. Based on
		identity, the client may be able to see additional information, such
		as membership specials.

		program_ids
		A list of program IDs on which to base the search.

		session_type_ids
		A list of session type IDs on which to base the search.

		location_ids
		A list of location IDs on which to base the search.

		semester_ids
		A list of semester IDs on which to base the search.

		hide_canceled_classes
		When `true`, canceled classes are removed from the response.<br />
		When `false`, canceled classes are included in the response.<br />
		Default: **false**

		scheduling_window
		When `true`, classes outside scheduling window are removed from the
		response.<br />  When `false`, classes are included in the response,
		regardless of the scheduling window.<br />  Default: **false**

		last_modified_date
		When included in the request, only records modified on or after the
		`LastModifiedDate` specified are included in the response.

		limit
		Number of results to include, defaults to 100

		offset
		Page offset, defaults to 0.
	"""

	input_map = {
		'ClassDescriptionIds': 'class_description_ids',
		'ClassIds': 'class_ids',
		'StaffIds': 'staff_ids',
		'StartDateTime': 'start_date_time',
		'EndDateTime': 'end_date_time',
		'ClientId': 'client_id',
		'ProgramIds': 'program_ids',
		'SessionTypeIds': 'session_type_ids',
		'LocationIds': 'location_ids',
		'SemesterIds': 'semester_ids',
		'HideCanceledClasses': 'hide_canceled_classes',
		'SchedulingWindow': 'scheduling_window',
		'LastModifiedDate': 'last_modified_date',
		'Limit': 'limit',
		'Offset': 'offset',
		}

	output_map = {
		'class_description_ids': 'ClassDescriptionIds',
		'class_ids': 'ClassIds',
		'staff_ids': 'StaffIds',
		'start_date_time': 'StartDateTime',
		'end_date_time': 'EndDateTime',
		'client_id': 'ClientId',
		'program_ids': 'ProgramIds',
		'session_type_ids': 'SessionTypeIds',
		'location_ids': 'LocationIds',
		'semester_ids': 'SemesterIds',
		'hide_canceled_classes': 'HideCanceledClasses',
		'scheduling_window': 'SchedulingWindow',
		'last_modified_date': 'LastModifiedDate',
		'limit': 'Limit',
		'offset': 'Offset',
		}

File: swagger/Models/GetClassesRequest.php
<?php

namespace App\Mindbody\Models;

/**
 * no description available
 */
class GetClassesRequest extends BaseModel
{
	/**
	 * The requested class description IDs.
	 * 
	 * @var integer[] 
	 */
	public $classDescriptionIds;
	/**
	 * The requested class IDs.
	 * 
	 * @var integer[] 
	 */
	public $classIds;
	/**
	 * The requested IDs of the teaching staff members.
	 * 
	 * @var integer[] 
	 */
	public $staffIds;
	/**
	 * The requested start date for filtering.   <br />Default: **today’s
	 * date**
	 * 
	 * @var string format:date-time
	 */
	public $startDateTime;
	/**
	 * The requested end date for filtering.  <br />Default: **today’s
	 * date**
	 * 
	 * @var string format:date-time
	 */
	public $endDateTime;
	/**
	 * The client ID of the client who is viewing this class list. Based on
	 * identity, the client may be able to see additional information, such
	 * as membership specials.
	 * 
	 * @var string 
	 */
	public $clientId;
	/**
	 * A list of program IDs on which to base the search.
	 * 
	 * @var integer[] 
	 */
	public $programIds;
	/**
	 * A list of session type IDs on which to base the search.
	 * 
	 * @var integer[] 
	 */
	public $sessionTypeIds;
	/**
	 * A list of location IDs on which to base the search.
	 * 
	 * @var integer[] 
	 */
	public $locationIds;
	/**
	 * A list of semester IDs on which to base the search.
	 * 
	 * @var integer[] 
	 */
	public $semesterIds;
	/**
	 * When `true`, canceled classes are removed from the response.<br />
	 * When `false`, canceled classes are included in the response.<br />
	 * Default: **false**
	 * 
	 * @var boolean 
	 */
	public $hideCanceledClasses;
	/**
	 * When `true`, classes outside scheduling window are removed from the
	 * response.<br />  When `false`, classes are included in the response,
	 * regardless of the scheduling window.<br />  Default: **false**
	 * 
	 * @var boolean 
	 */
	public $schedulingWindow;
	/**
	 * When included in the request, only records modified on or after the
	 * `LastModifiedDate` specified are included in the response.
	 * 
	 * @var string format:date-time
	 */
	public $lastModifiedDate;
	/**
	 * Number of results to include, defaults to 100
	 * 
	 * @var integer format:int32
	 */
	public $limit;
	/**
	 * Page offset, defaults to 0.
	 * 
	 * @var integer format:int32
	 */
	public $offset;

	protected function getInputMap()
	{
		return [
			'ClassDescriptionIds' => ['classDescriptionIds', 'array', null],
			'ClassIds' => ['classIds', 'array', null],
			'StaffIds' => ['staffIds', 'array', null],
			'StartDateTime' => ['startDateTime', 'string', 'date-time'],
			'EndDateTime' => ['endDateTime', 'string', 'date-time'],
			'ClientId' => ['clientId', 'string', null],
			'ProgramIds' => ['programIds', 'array', null],
			'SessionTypeIds' => ['sessionTypeIds', 'array', null],
			'LocationIds' => ['locationIds', 'array', null],
			'SemesterIds' => ['semesterIds', 'array', null],
			'HideCanceledClasses' => ['hideCanceledClasses', 'boolean', null],
			'SchedulingWindow' => ['schedulingWindow', 'boolean', null],
			'LastModifiedDate' => ['lastModifiedDate', 'string', 'date-time'],
			'Limit' => ['limit', 'integer', 'int32'],
			'Offset' => ['offset', 'integer', 'int32'],
			];
	}

	protected function getOutputMap()
	{
		return [
			'classDescriptionIds' => ['ClassDescriptionIds', 'array', null],
			'classIds' => ['ClassIds', 'array', null],
			'staffIds' => ['StaffIds', 'array', null],
			'startDateTime' => ['StartDateTime', 'string', 'date-time'],
			'endDateTime' => ['EndDateTime', 'string', 'date-time'],
			'clientId' => ['ClientId', 'string', null],
			'programIds' => ['ProgramIds', 'array', null],
			'sessionTypeIds' => ['SessionTypeIds', 'array', null],
			'locationIds' => ['LocationIds', 'array', null],
			'semesterIds' => ['SemesterIds', 'array', null],
			'hideCanceledClasses' => ['HideCanceledClasses', 'boolean', null],
			'schedulingWindow' => ['SchedulingWindow', 'boolean', null],
			'lastModifiedDate' => ['LastModifiedDate', 'string', 'date-time'],
			'limit' => ['Limit', 'integer', 'int32'],
			'offset' => ['Offset', 'integer', 'int32'],
			];
	}
}
